### ISSUE TITLE
Create feature flag for enabling expandPartialFromOptionToNonOption

### ISSUE THREAD

**Issue by olivierdeckers** [scalalandio/chimney#322]
TLDR: The user should explicitly enable a feature flag to enable derivation using `expandPartialFromOptionToNonOption`. 

In the current situation, a user has two options: 
* use Transformer, the transformation cannot fail (in a way where problems are aggregated)
* use PartialTransformer, transformation can fail and problems are aggregated. However, Option[T] -> T partial transformer is derived automatically, so no compilation error if this is not what the user meant.

It would be nice to have some middle ground, where if we don't explicitly enable a feature flag, the following results in a compilation error:

```
val x: Option[T] = None
...
val y = x.transformIntoPartial[T].asEither // Left(empty)
```

This may seem obvious in a simple example, but in more complicated code while refactoring if let's say `x` is changed from `T` to `Option[T]`, the compiler no longer makes you think about maybe `y` should be `Option[T]` as well.

**Comment by MateuszKubuszok** [comment]
I added the flag allowing to disable the behavior, however I do not intend to make that behavor disabled by default as it would make Partials to much PITA to use most of the time. I explained my reasons in #477 in more details. The flag is released and available since 1.0.0-M2.
